libraries/botframework-streaming/src/namedPipe/namedPipeTransport.ts

```diff
@@ -85,7 +85,11 @@ export class NamedPipeTransport implements ITransportSender, ITransportReceiver
      * @returns The buffer containing the data from the transport.
      */
     receive(count: number): Promise<INodeBuffer> {
-        if (this._activeReceiveResolve) {
+        if (!this.socket) {
+            throw new Error('Socket unavailable to make the connection.');
+        } else if (this.socket.destroyed) {
+            throw new Error('Socket was destroyed.');
+        } else if (this._activeReceiveResolve) {

```

Comment:
These are other error messages i think of.
throw new Error('Cannot receive data over an unavailable/null socket.');
throw new Error('Cannot receive data over a dead/destroyed socket.');